### ISSUE TITLE
Support for link seam mocking of non-virtual class members

### ISSUE THREAD

**Issue by Hawk777** [rollbear/trompeloeil#156]
I wanted to use Trompeloeil for link seam mocking of non-virtual class member functions. That is, assume I have an existing header file `MyClass.h`:
```c++
class MyClass {
  public:
  void f();
};
```

I have a real-world implementation in `MyClass.cpp`:
```c++
#include "MyClass.h"
void MyClass::f() {
  // Do real work
}
```

I want to unit test something that uses a MyClass, and I expected to be able to do it in a way vaguely like the following, where I would avoid multiple definitions by *not linking* `MyClass.o` when building the unit tests:
```c++
#include "MyClass.h"

MAKE_MOCK0(MyClass::f, void());

int main() {
  MyClass obj;
  REQUIRE_CALL(obj, f());
  obj.f();
  return 0;
}
```

But it seems that the `MAKE_MOCK0` macro expands to stuff that must live within the class body, which sadly makes this impossible. Is there any chance of this being supported?

**Comment by rollbear** [comment]
I understand. I don't see how this can be done, because the macro really needs to add things to the class. That is very fundamental to how Trompeloeil works (the list of current and expired expectations are per-instance data, for example.)

**Comment by Hawk777** [comment]
Yeah, I looked at the definition of the macro and figured this would be a pretty big refactor. I got really excited when I saw [this](https://github.com/rollbear/trompeloeil/blob/master/docs/CookBook.md#mocking_non_virtual) and then got disappointed when I realized it meant mocking non-virtual members in a freshly created class, not mocking non-virtual members in an existing class. Trompeloeil probably isn’t the right choice for me; feel free to leave this open or close as WONTFIX as you prefer.
